### ISSUE TITLE
Subclasses of abstract class that do not inherit abc.ABC are considered abstract

### ISSUE THREAD

**Issue by sshane** [PyCQA/pylint#7950]
### Bug description

Related to https://github.com/PyCQA/pylint/issues/3098. Subclasses of an abstract parent class which do not inherit directly from `abc.ABC` are considered abstract, and only throw an error during run time when trying to instantiate it.

For example, take this code below. PyCharm correctly determines through a IDE warning that Sub is not abstract, and says `Class Sub must implement all abstract methods`, but pylint will ignore this class, as its ancestors inherit from `abc.ABC`. Do we not get the best of both worlds (allowing abstract subclasses of abstract parent classes, ie. `AbsSub`, while catching non-abstract subclasses of abstract parent classes, ie. `Sub`) if we check the bases, instead of the ancestors where that PR to fix the issue above does it?

```python
import abc
import unittest

class Abstract(unittest.TestCase, abc.ABC):
  @abc.abstractmethod
  def test_something(self):
    pass

  @abc.abstractmethod
  def test_another_thing(self):
    pass


class AbsSub(Abstract, abc.ABC):
  def test_another_thing(self):
    return 1


class Sub(AbsSub):
  pass
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
`NOTHING`
```

### Expected behavior

`test.py:19:0: W0223: Method 'test_something' is abstract in class 'Abstract' but is not overridden (abstract-method)` (referring to `Sub`)

### Pylint version

```shell
2.5.0
```

### OS / Environment

Ubuntu 20.04

**Comment by DanielNoord** [comment]
Seems like the check is indeed incorrect!

**Comment by sshane** [comment]
If not one wants to/has time to open a PR, I can try to tackle it.

**Comment by DanielNoord** [comment]
That would definitely be appreciated!

**Comment by DetachHead** [comment]
as a workaround, you can define your own version of `ABC` instead:
```py
from abc import abstractmethod
import unittest

class ABC(metaclass=ABCMeta):
    pass

class Abstract(unittest.TestCase, ABC):
  @abstractmethod
  def test_something(self):
    pass

  @abstractmethod
  def test_another_thing(self):
    pass


class AbsSub(Abstract, abc.ABC):
  def test_another_thing(self):
    return 1


class Sub(AbsSub):  # error: Method 'test_something' is abstract in class 'Abstract' but is not overridden in child class 'Sub'
  pass
```
